components/text_sensor/modbus_controller.rst

```diff
@@ -58,55 +56,24 @@ Possible return values for the lambda:
-        id: template_rtc
-
-    - platform: modbus_controller
-        modbus_controller_id: traceranx
-        name: "rtc clock test"
-        id: rtc_clock_test
-        internal: true
+      - platform: modbus_controller
+        modbus_controller_id: modbus_device
+        id: reg_1002_text
+        bitmask: 0
         register_type: holding
-        address: 0x9013
-        register_count: 3
-        hex_encode: true
-        response_size: 6
-        on_value:
-            then:
-            - lambda: |-
-                ESP_LOGV("main", "decoding rtc hex encoded raw data: %s", x.c_str());
-                uint8_t h=0,m=0,s=0,d=0,month_=0,y = 0 ;
-                m = esphome::modbus_controller::byte_from_hex_str(x,0);
-                s = esphome::modbus_controller::byte_from_hex_str(x,1);
-                d = esphome::modbus_controller::byte_from_hex_str(x,2);
-                h = esphome::modbus_controller::byte_from_hex_str(x,3);
-                y = esphome::modbus_controller::byte_from_hex_str(x,4);
-                month_ = esphome::modbus_controller::byte_from_hex_str(x,5);
-                // Now check if the rtc time of the controller is ok and correct it
-                time_t now = ::time(nullptr);
-                struct tm *time_info = ::localtime(&now);
-                int seconds = time_info->tm_sec;
-                int minutes = time_info->tm_min;
-                int hour = time_info->tm_hour;
-                int day = time_info->tm_mday;
-                int month = time_info->tm_mon + 1;
-                int year = time_info->tm_year - 2000;
-                // correct time if needed (ignore seconds)
-                if (d != day || month_ != month || y != year || h != hour || m != minutes) {
-                    // create the payload
-                    std::vector<uint16_t> rtc_data = {uint16_t((minutes << 8) | seconds), uint16_t((day << 8) | hour),
-                                                    uint16_t((year << 8) | month)};
-                    // Create a modbus command item with the time information as the payload
-                    esphome::modbus_controller::ModbusCommandItem set_rtc_command = esphome::modbus_controller::ModbusCommandItem::create_write_multiple_command(traceranx, 0x9013, 3, rtc_data);
-                    // Submit the command to the send queue
-                    traceranx->queue_command(set_rtc_command);
-                    ESP_LOGI("ModbusLambda", "EPSOLAR RTC set to %02d:%02d:%02d %02d.%02d.%04d", hour, minutes, seconds, day, month, year + 2000);
-                }
-                char buffer[20];
-                // format time as YYYY:mm:dd hh:mm:ss
-                sprintf(buffer,"%04d:%02d:%02d %02d:%02d:%02d",y+2000,month_,d,h,m,s);
-                id(template_rtc).publish_state(buffer);
+        address: 1002
+        raw_encode: HEXBYTES
+        name: Register 1002 (Text)
+        lambda: |-
+          uint16_t value = modbus_controller::word_from_hex_str(x, 0);
```

Comment:
I like how this shorter lambda really lets you focus on the rest of the configuration more easily.